app/controllers/users_controller.rb

```diff
@@ -23,9 +28,56 @@ def create
     end
   end
 
+  def edit
+    find_user
+  end
+
+  def update
+    find_user
+    if @user.update user_params
+      flash[:success] = t ".profile_updated"
+      redirect_to @user
+    else
+      flash[:danger] = t ".update_failed"
+      render :edit
+    end
+  end
+
+  def destroy
+    @user.destroy
+    flash[:success] = t ".user_deleted"
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit User::USERS_PARAMS
   end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t ".please_log_in"
+      redirect_to login_url
+    end
```

Comment:
Các anh chị sẽ bảo bạn chuyển thành như này đó, convention cũng có đó ạ, nên dùng guard clause thay cho việc wrap code trong điều kiện...

```suggestion
return if logged_in?

store_location
flash[:danger] = t ".please_log_in"
redirect_to login_url
```